File: solutions/Leetcode_415/leetcode_415.py
class Solution1:
    def add_string(self, num1: str, num2: str) -> str:
        ret = []
        add_one = 0
        if len(num1) > len(num2):
            num2 = '0' * (len(num1)-len(num2)) + num2
        elif len(num1) < len(num2):
            num1 = '0' * (len(num2)-len(num1)) + num1
        for i in range(len(num1)-1, -1, -1):
            x = (ord(num1[i])-48) + (ord(num2[i])-48) + add_one
            add_one = x // 10
            x = str(x % 10)
            ret.append(x)
        if add_one:
            ret.append('1')
        return ''.join(ret[::-1])

File: solutions/Leetcode_415/leetcode_415.cpp
#include <algorithm>
#include <string>

class Solution1
{
public:
    std::string addStrings(std::string num1, std::string num2)
    {
        int i = num1.size() - 1;
        int j = num2.size() - 1;
        std::string ret = "";
        int addOne = 0;
        while (i >= 0 || j >= 0 || addOne)
        {
            int x1 = i >= 0 ? num1[i] - '0' : 0;
            int x2 = j >= 0 ? num2[j] - '0' : 0;
            i--;
            j--;
            int sum = x1 + x2 + addOne;
            addOne = sum / 10;
            sum %= 10;
            ret.push_back('0' + sum);
        }
        std::reverse(ret.begin(), ret.end());
        return ret;
    }
};